include/aws/common/uri.h

```diff
@@ -114,6 +114,30 @@ AWS_COMMON_API uint16_t aws_uri_port(const struct aws_uri *uri);
  */
 AWS_COMMON_API const struct aws_byte_cursor *aws_uri_path_and_query(const struct aws_uri *uri);
 
+/**
+ * For iterating over the params in the query string.
+ * `param` is an in/out argument used to track progress, it MUST be zeroed out to start.
+ * If true is returned, `param` contains the value of the next param.
+ * If false is returned, there are no further params.
+ *
+ * Edge cases:
+ * 1) Entries without '=' sign are treated as having a key and no value.
+ *    Example: First param in query string "a&b=c" has key="a" value=""
+ *
+ * 2) Blank entries are skipped.
+ *    Example: The only param in query string "&&a=b" is key="a" value="b"
+ */
+AWS_COMMON_API bool aws_query_string_next_param(
+    const struct aws_byte_cursor *query_string,
```

Comment:
I know it's pretty inconsistent in our codebase, but cursors by-value are clearly superior
```suggestion
   struct aws_byte_cursor query_string,
```